packages/vscode_simple_ext/src/logger.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * This file manages the logger's state.
+ */
+import { readFileSync } from "fs";
+import { resolve } from "path";
+import { ok } from "assert";
+import { ExtensionContext, window, workspace } from "vscode";
+import { IChildLogger, IVSCodeExtLogger } from "@vscode-logging/types";
+import { configureLogger, NOOP_LOGGER } from "@vscode-logging/wrapper";
+import { LOGGING_LEVEL_PROP, SOURCE_LOCATION_PROP} from "./constants"
+
+// On file load we initialize our logger to `NOOP_LOGGER`
+// this is done because the "real" logger cannot be initialized during file load.
+// only once the `activate` function has been called in extension.ts
+// as the `ExtensionContext` argument to `activate` contains the required `logPath`
+let loggerImpel: IVSCodeExtLogger = NOOP_LOGGER;
+
+export function getLogger(): IChildLogger {
+  return loggerImpel;
+}
+
+function setLogger(newLogger: IVSCodeExtLogger): void {
+  loggerImpel = newLogger;
+}
+
+export function initLogger(context: ExtensionContext): void {
+  const meta = JSON.parse(
+    readFileSync(resolve(context.extensionPath, "package.json"), "utf8")
+  );
+
+  const extLogger = configureLogger({
+    extName: meta.displayName,
+    logPath: context.logPath,
+    logOutputChannel: window.createOutputChannel(meta.displayName),
+    // set to `true` if you also want your VSCode extension to log to the console.
+    logConsole: false,
```

Comment:
What is the recommendation here? Since people are going to copy this code it should contain the recommended approach (maybe with an explanation) instead of instructions that are supposed to make people think

Reply:
Honestly I am not sure what is the recommendation for internal usage now.
Lets forward this question to @saggir 

It can be changed after merge by @saggir if needed.